File: functions/image-processing/handler.py
from os import path
from time import time
from PIL import Image
import json

from .ops import *

SCRIPT_DIR = path.abspath(path.join(path.dirname(__file__)))
IMAGE_DIR = path.join(SCRIPT_DIR, 'images')
IMAGE_PATH = path.join(IMAGE_DIR, 'image.jpg')
IMAGE_NAME = 'image.jpg'

def handle(event, context):
    try:
        req = json.loads(event.body.decode())
    except Exception as _:
        image_id = 0
    else:
        image_id = req.get('image_id', 0)
    start = time()
    if image_id == 0:
        IMAGE_PATH = path.join(IMAGE_DIR, 'image.jpg')
    else:
        IMAGE_PATH = path.join(IMAGE_DIR, 'image2.jpg')
    path_list = []
    with Image.open(IMAGE_PATH) as image:
        path_list += flip(image, IMAGE_NAME)
        path_list += rotate(image, IMAGE_NAME)
        path_list += filter(image, IMAGE_NAME)
        path_list += gray_scale(image, IMAGE_NAME)
        path_list += resize(image, IMAGE_NAME)

    latency = time() - start
    return {
        "statusCode": 200,
        "body": {
            'latency': latency,
            'data': path_list
        }
    }

File: functions/image-flip-rotate/handler.js
"use strict";

const path = require("path");
const sharp = require("sharp");

module.exports = async (event, context) => {
  let image_id = 0;
  try {
    const req = typeof event.body === "object" ? event.body : JSON.parse(event.body);
    if ("image_id" in req) {
      image_id = req.image_id;
    }
  } catch {}
  const start = Date.now();
  let p;
  if (image_id == 0) {
    p = path.join(__dirname, "images", "image.jpg");
  } else {
    p = path.join(__dirname, "images", "image2.jpg");
  }

  const image = sharp(p);
  await Promise.all([
    image.flip().toFile("/run/flip-top-bottom.jpg"),
    image.flop().toFile("/run/flop-left-right.jpg"),
    image.rotate(90).toFile("/run/rotate-90.jpg"),
    image.rotate(180).toFile("/run/rotate-180.jpg"),
    image.rotate(270).toFile("/run/rotate-270.jpg"),
  ]);

  const latency = Date.now() - start;

  return context
    .status(200)
    .succeed({
      latency: latency / 1000,
    });
};
